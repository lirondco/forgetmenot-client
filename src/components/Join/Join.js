import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import AuthApiService from '../../services/auth-api-service'

//renders the registration page

export default class Join extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = ev => {
    //function to handle new user registrations
    ev.preventDefault()
    const { username, email, password } = ev.target

    this.setState({ error: null })
    AuthApiService.postUser({
      username: username.value,
      email: email.value,
      password: password.value,
    })
      .then(user => {
        username.value = ''
        email.value = ''
        password.value = ''
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    return (
      <form
        className='RegistrationForm'
        onSubmit={this.handleSubmit}
      >
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='username'>
          <label htmlFor='RegistrationForm__username'>
            Username <Required />
          </label>
          <Input
            name='username'
            type='text'
            required
            id='RegistrationForm__username'>
          </Input>
        </div>
        <div className='email'>
          <label htmlFor='RegistrationForm__email'>
            Email Address <Required />
          </label>
          <Input
            name='email'
            type='email'
            required
            id='RegistrationForm__email'>
          </Input>
        </div>
        <div className='password'>
          <label htmlFor='RegistrationForm__password'>
            Password <Required />
          </label>
          <Input
            name='password'
            type='password'
            required
            id='RegistrationForm__password'>
          </Input>
        </div>
        <Button type='submit'>
          Register
        </Button>
      </form>
    )
  }
}