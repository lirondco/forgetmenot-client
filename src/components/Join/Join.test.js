import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Join from './Join'

describe(`Join component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<Join />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})