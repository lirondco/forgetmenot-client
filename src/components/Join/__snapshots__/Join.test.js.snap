// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join component renders the complete form 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    className="username"
  >
    <label
      htmlFor="RegistrationForm__username"
    >
      Username 
      <Required />
    </label>
    <Input
      id="RegistrationForm__username"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <div
    className="email"
  >
    <label
      htmlFor="RegistrationForm__email"
    >
      Email Address 
      <Required />
    </label>
    <Input
      id="RegistrationForm__email"
      name="email"
      required={true}
      type="email"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password 
      <Required />
    </label>
    <Input
      id="RegistrationForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <Button
    type="submit"
  >
    Register
  </Button>
  <div
    role="alert"
  />
</form>
`;
